src/language-js/needs-parens.js

```diff
@@ -5,19 +5,39 @@ const assert = require("assert");
 const util = require("../common/util");
 const comments = require("./comments");
 
-function hasClosureCompilerTypeCastComment(text, node, locEnd) {
+function hasClosureCompilerTypeCastComment(text, path, locStart, locEnd) {
```

Comment:
How do you feel about dropping `ClosureCompiler` from this name since it supports TS too?

Reply:
Actually, that syntax seems coming from closure compiler, see Microsoft/TypeScript#5158.

Reply:
Hi is this the correct place to comment.

Reply:
Hi is this the correct place to comment.

Reply:
Salsa ≠ TypeScript (the language)

Reply:
What’s Salsa then?

Reply:
Ah, [TS support for JS](https://github.com/Microsoft/TypeScript/issues/4789).